event-svc/src/event/service.rs

```diff
@@ -304,17 +306,29 @@ impl EventService {
 
                 self.notify_ordering_task(&store_result).await?;
 
-                Ok(InsertResult::new_from_store(invalid, store_result))
+                Ok(Self::partition_store_result(store_result))
             }
             DeliverableRequirement::Lazy => {
                 let store_result =
                     CeramicOneEvent::insert_many(&self.pool, to_insert.iter()).await?;
 
-                Ok(InsertResult::new_from_store(invalid, store_result))
+                Ok(Self::partition_store_result(store_result))
             }
         }
     }
 
+    /// Returns two vectors of Event IDs representing (new, existed)
+    fn partition_store_result(store: crate::store::InsertResult) -> (Vec<EventId>, Vec<EventId>) {
+        store.inserted.into_iter().partition_map(|e| {
+            let key = e.inserted.order_key().clone();
+            if e.new_key {
+                itertools::Either::Left(key)
```

Comment:
Why are we using an Either rather than a more meaningful enum, where we can name left and right.

Reply:
that's the signature of the Itertools [partition_map](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.partition_map) function.. I could make a struct return instead of the two vecs, but it was just a small internal helper and we don't need more info yet so I just used the vecs with `partition_map` to help split them